mount/mount_unix.go

```diff
@@ -61,12 +61,18 @@ func RecursiveUnmount(target string) error {
 		return len(mounts[i].Mountpoint) > len(mounts[j].Mountpoint)
 	})
 
-	var suberr error
+	var (
+		suberr    error
+		lastMount = len(mounts) - 1
```

Comment:
FWIW, not sure how smart the compiler would be, and if this change would be a no-op; if so, I can drop this commit.

Reply:
I think `len()` is a very fast operation in Go, as it just returns the length from the internal structure that represents a slice.

A benchmark would be helpful to find out for sure

Reply:
The difference is negligible (63 vs 64 ns per run), but it makes the code more readable so I think we can keep it.

Test case:
```go
package mountinfo

import "testing"

func BenchmarkLenInLoop(b *testing.B) {
	mounts, err := GetMounts(nil)
	if err != nil {
		b.Fatal(err)
	}

	var last *Info
	b.ResetTimer()
	for n := 0; n < b.N; n++ {
		for i, m := range mounts {
			if i == len(mounts)-1 { // last mount
				last = m
			}
		}
	}
	b.Log("Last mount " + last.Mountpoint)
}

func BenchmarkLenOutOfLoop(b *testing.B) {
	mounts, err := GetMounts(nil)
	if err != nil {
		b.Fatal(err)
	}

	var (
		last         *Info
		lastMountIdx = len(mounts) - 1
	)
	b.ResetTimer()
	for n := 0; n < b.N; n++ {
		for i, m := range mounts {
			if i == lastMountIdx { // last mount
				last = m
			}
		}
	}
	b.Log("Last mount " + last.Mountpoint)
}
```

Results:
``console
$ go test -bench Len -benchtime 10s .
goos: linux
goarch: amd64
pkg: github.com/moby/sys/mountinfo
BenchmarkLenInLoop-8      	186409401	        64.3 ns/op
--- BENCH: BenchmarkLenInLoop-8
    bench_test.go:20: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
    bench_test.go:20: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
    bench_test.go:20: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
    bench_test.go:20: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
    bench_test.go:20: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
    bench_test.go:20: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
BenchmarkLenOutOfLoop-8   	191245992	        63.2 ns/op
--- BENCH: BenchmarkLenOutOfLoop-8
    bench_test.go:41: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
    bench_test.go:41: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
    bench_test.go:41: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
    bench_test.go:41: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
    bench_test.go:41: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
    bench_test.go:41: Last mount /run/media/kir/8885b6f8-2cc0-4416-b5e9-dfecb3137d9a
PASS
ok  	github.com/moby/sys/mountinfo	36.921s
```

Reply:
Ah, yes, that's not much of a difference; thanks for testing!